### DISCUSSION TITLE
How to dynamically rerender a image

### DISCUSSION THREAD

**Discussion by UnknownSuperficialNight** [emilk/egui#4253] [Q&A] [answered]
I'm trying to load an image, and I did it successfully via this:
`
ui.image("file:///mysources/Rust/rust-gui/target/x86_64-unknown-linux-gnu/release/Music_app_config/image.png");`

I'm using the following in cargo.toml:

`egui_extras = { version = "*", features = ["all_loaders"] }`

This works great, but if, for example, I delete the image while the program is running and replace it with a new one, it does not dynamically re-render with the new image. Is there a way to force it to re-render the image?

**Comment by YgorSouza** [comment, marked as answer]
You have to call [Context::forget_image](https://docs.rs/egui/0.27.0/egui/struct.Context.html#method.forget_image) when you want the image to be reloaded. Note that doing that on every frame might increase the CPU usage significantly, so you might want to add some check that the file has changed, such as checking the [modified](https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.modified) timestamp. The default egui texture loader does not do this automatically, but if you want you can also create your own loaders with this functionality so you don't have to do it everywhere in the code. I talked about that in <https://github.com/emilk/egui/issues/3501#issuecomment-1958822073>.
